zebra/zserv.c

```diff
@@ -2108,6 +2109,68 @@ static void zread_mpls_labels(ZAPI_HANDLER_ARGS)
 	return;
 }
 
+static int zsend_table_manager_connect_response(struct zserv *client,
+						vrf_id_t vrf_id, u_short result)
+{
+	int ret;
+	struct stream *s = stream_new(ZEBRA_MAX_PACKET_SIZ);
+
+	zclient_create_header(s, ZEBRA_TABLE_MANAGER_CONNECT, vrf_id);
+
+	/* result */
+	stream_putc(s, result);
+
+	/* Write packet size. */
+	stream_putw_at(s, 0, stream_get_endp(s));
+
+	ret = writen(client->sock, s->data, stream_get_endp(s));
+	stream_free(s);
+
+	return ret;
+}
+
+/* Send response to a table manager connect request to client */
+static void zread_table_manager_connect(struct zserv *client, struct stream *msg,
+					vrf_id_t vrf_id)
+{
+	struct stream *s;
+	u_char proto;
+	u_short instance;
+
+	s = msg;
+
+	/* Get data. */
+	STREAM_GETC(s, proto);
+	STREAM_GETW(s, instance);
+
+	/* accept only dynamic routing protocols */
+	if ((proto >= ZEBRA_ROUTE_MAX) || (proto <= ZEBRA_ROUTE_STATIC)) {
+		zlog_err("client %d has wrong protocol %s", client->sock,
+			 zebra_route_string(proto));
+		zsend_table_manager_connect_response(client, vrf_id, 1);
+		return;
+	}
+	zlog_notice("client %d with instance %u connected as %s", client->sock,
+		    instance, zebra_route_string(proto));
+	client->proto = proto;
+	client->instance = instance;
+
+	/*
+	 * Release previous labels of same protocol and instance.
+	 * This is done in case it restarted from an unexpected shutdown.
+	 */
+	release_daemon_table_chunks(proto, instance);
+
+	zlog_debug(
```

Comment:
unguarded debug?

Reply:
I did duplicate from label maanger, so was thinkning it would be acceptable.
I think debug is useless here, so I remove it